client/test/unit/lib/solana/transactions/expandTransaction.test.ts

```diff
@@ -0,0 +1,148 @@
+import * as sinon from 'sinon';
+
+import { Keypair, PublicKey } from '@solana/web3.js';
+import {
+  deriveDefaultDOAFromKey,
+  didToPDA,
+  publicKeyToDid,
+} from '../../../../../src/lib/solana/util';
+import { stubGetBlockhash } from '../../../../utils/lang';
+import { create } from '../../../../../src/lib/solana/instructions/expandTransaction';
+import { range } from 'ramda';
+import { randomInt } from 'crypto';
+import { AccountOperation } from '../../../../../src/lib/solana/model/AccountOperation';
+import { AssignablePublicKey } from '../../../../../src/lib/solana/model/AssignablePublicKey';
+import { UnitValue } from '../../../../../src/lib/solana/solanaBorsh';
+import { normalizeSigner } from '../../../../../src/lib/util';
+import {
+  AddAccount,
+  AddAccounts,
+  AddData,
+  InstructionOperation,
+} from '../../../../../src/lib/solana/model/InstructionOperation';
+import { InstructionData } from '../../../../../src/lib/solana/model/InstructionData';
+import { TransactionAccountMeta } from '../../../../../src/lib/solana/model/TransactionAccountMeta';
+
+const sandbox = sinon.createSandbox();
+
+describe('transactions/expandTransaction', function () {
+  const payer = Keypair.generate();
+  const did = publicKeyToDid(payer.publicKey);
+  let didPDAKey: PublicKey;
+
+  beforeEach(() => stubGetBlockhash(sandbox));
+  afterEach(sandbox.restore);
+
+  before(async () => {
+    didPDAKey = await didToPDA(did);
+  });
+
+  it('should create an expand instruction', async () => {
+    const accountOperations = range(0, randomInt(2, 10)).map(() => {
+      switch (randomInt(0, 3)) {
+        case 0:
+          return new AccountOperation({
+            add: AssignablePublicKey.fromPublicKey(
+              Keypair.generate().publicKey
+            ),
+          });
+        case 1:
+          return new AccountOperation({
+            clear: new UnitValue(),
+          });
+        case 2:
+          return new AccountOperation({
+            addMany: range(0, randomInt(2, 10)).map(() =>
+              AssignablePublicKey.fromPublicKey(Keypair.generate().publicKey)
+            ),
+          });
+        default:
+          throw new Error('Out of bounds');
+      }
+    });
+
+    const instructionOperations: InstructionOperation[] = range(
+      0,
+      randomInt(2, 10)
+    ).map(() => {
```

Comment:
See above - also, even if sticking with a pattern like this, it needs to be pulled out into a util module of some kind as it is too large for a test